leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java

```diff
@@ -190,35 +221,89 @@ public static void createAndStartServer(int webPort, String localAddress, int lo
-
-            // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
-
-            // Get keys
-            publicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);
-            PrivateKey privateKey = KeyFactory.getInstance("EC").generatePrivate(privateKeySpec);
-            builder.setPublicKey(publicKey);
-            builder.setPrivateKey(privateKey);
-        } catch (InvalidKeySpecException | NoSuchAlgorithmException | InvalidParameterSpecException e) {
-            LOG.error("Unable to initialize RPK.", e);
-            System.exit(-1);
+
+        // Set up X.509 mode
+        if (keyStorePath != null) {
+            try {
+                KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+                try (FileInputStream fis = new FileInputStream(keyStorePath)) {
+                    keyStore.load(fis, keyStorePass == null ? null : keyStorePass.toCharArray());
+                    ArrayList<Certificate> trustedCertificates = new ArrayList<>();
+                    for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();) {
+                        String alias = aliases.nextElement();
+                        if (keyStore.isCertificateEntry(alias)) {
+                            trustedCertificates.add(keyStore.getCertificate(alias));
+                        } else if (keyStore.isKeyEntry(alias) && alias.equals(keyStoreAlias)) {
+                            ArrayList<X509Certificate> x509CertificateChain = new ArrayList<>();
+                            Certificate[] certificateChain = keyStore.getCertificateChain(alias);
+                            if (certificateChain != null) {
+                                for (Certificate certificate : certificateChain) {
+                                    if (certificate instanceof X509Certificate) {
+                                        x509CertificateChain.add((X509Certificate) certificate);
+                                    } else {
+                                        LOG.warn("Ignoring non-X.509 certificate in alias chain: {}", certificate);
+                                    }
+                                }
+                            }
+                            if (!x509CertificateChain.isEmpty()) {
+                                Key key = keyStore.getKey(alias,
+                                        keyStoreAliasPass == null ? new char[0] : keyStoreAliasPass.toCharArray());
+                                if (key instanceof PrivateKey) {
+                                    builder.setPrivateKey((PrivateKey) key);
+                                    publicKey = keyStore.getCertificate(alias).getPublicKey();
+                                    builder.setPublicKey(publicKey);
```

Comment:
You don't need set public key as explain in LeshanServer javadoc. We should add javadoc to builder too.